test/dart/devtools_integration/_envelopes_test.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:leak_tracker/src/devtools_integration/_envelopes.dart';
+import 'package:test/test.dart';
+
+import '../../test_infra/data/messages.dart';
+
+void main() {
+  setUp(() {
+    verifyTestsCoverAllEnvelopes();
+  });
+
+  test('each code matches exactly one envelope', () {
+    final codesInEnvelopes = Set.from(envelopes.map((e) => e.code));
+    expect(codesInEnvelopes, hasLength(Codes.values.length));
+  });
+
+  test('envelopes are unique by type', () {
+    final types = Set.from(envelopes.map((e) => e.type));
+    expect(types, hasLength(envelopes.length));
+  });
+
+  for (final message in messages) {
+    test('envelopes preserve original type', () {
```

Comment:
```suggestion
    test('envelopes preserve $runtimeType type', () {
```
When debugging failing tests it can be helpful to include information about which case is being tested, in the name